numpy/core/src/umath/ufunc_object.c

```diff
@@ -5079,14 +5079,17 @@ ufunc_generic_vectorcall(PyObject *ufunc,
 
 
 NPY_NO_EXPORT PyObject *
-ufunc_geterr(PyObject *NPY_UNUSED(dummy), PyObject *args)
+ufunc_geterr(PyObject *NPY_UNUSED(dummy), PyObject *const *args, Py_ssize_t len_args)
 {
     PyObject *thedict;
     PyObject *res;
 
-    if (!PyArg_ParseTuple(args, "")) {
+    NPY_PREPARE_ARGPARSER;
+    if (npy_parse_arguments("geterrobj", args, len_args, NULL,
+                NULL, NULL, NULL) < 0) {
```

Comment:
Yeah, I really think here using `METH_NOARGS` would be nice... saves us the whole parsing stuff after all.

Reply:
Yes, looks much cleaner then. Also performance goes from about 75 to 70 ns.